crates/bevy_ui/src/render/mod.rs

```diff
@@ -522,125 +529,72 @@ pub fn extract_uinode_borders(
-                inner_rect.min.x,
-                outer_rect.min.y,
-                inner_rect.max.x,
-                inner_rect.min.y,
-            ),
-            // Bottom edge
-            Rect::new(
-                inner_rect.min.x,
-                inner_rect.max.y,
-                inner_rect.max.x,
-                outer_rect.max.y,
-            ),
-        ];
-
-        let world_from_local = global_transform.compute_matrix();
-        for edge in outline_edges {
-            if edge.min.x < edge.max.x && edge.min.y < edge.max.y {
+            if let Some(border_color) = maybe_border_color {
+                let transform = global_transform.compute_matrix();
                 extracted_uinodes.uinodes.insert(
                     commands.spawn_empty().id(),
                     ExtractedUiNode {
-                        stack_index: node.stack_index,
+                        stack_index: uinode.stack_index,
                         // This translates the uinode's transform to the center of the current border rectangle
-                        transform: world_from_local
-                            * Mat4::from_translation(edge.center().extend(0.)),
-                        color: outline.color.into(),
+                        transform,
+                        color: border_color.0.into(),
                         rect: Rect {
-                            max: edge.size(),
+                            max: uinode.size(),
                             ..Default::default()
                         },
                         image,
                         atlas_scaling: None,
-                        clip: maybe_clip.map(|clip| clip.clip),
+                        clip: clip.map(|clip| clip.clip),
                         flip_x: false,
                         flip_y: false,
                         camera_entity,
-                        border: [0.; 4],
-                        border_radius: [0.; 4],
-                        node_type: NodeType::Rect,
+                        border_radius,
+                        border,
+                        node_type: NodeType::Border,
                     },
                 );
             }
         }
+
+        if let Some(outline) = maybe_outline {
+            let outer_distance = uinode.outline_offset() + uinode.outline_width();
+            let outline_radius = border_radius.map(|radius| {
+                if 0. < radius {
+                    radius + outer_distance
+                } else {
+                    0.
+                }
+            });
+            let outline_size = uinode.size() + 2. * outer_distance;
+            extracted_uinodes.uinodes.insert(
+                commands.spawn_empty().id(),
+                ExtractedUiNode {
+                    stack_index: uinode.stack_index,
+                    // This translates the uinode's transform to the center of the current border rectangle
+                    transform: global_transform.compute_matrix(),
```

Comment:
Could you use a descriptive variable here instead of the comment? I know this is pre-existing code, but would be nice to clean this up a little bit while we are on it.

Reply:
Now that the borders aren't assembled from multiple rectangles the comment no longer makes sense so I deleted it.